core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -2696,16 +2696,6 @@ protected void convertCollectionTable(
             validator().getValidatedNodeType(call),
             columnMappings);
 
-    final SqlValidatorScope selectScope =
```

Comment:
this snippet were introduced by a [1] and addresses the very particular case with LogicalTableFunctionScan relation.

Current patch should solve the problem in general, thus this snippet is not required anymore

[1] https://github.com/apache/calcite/pull/2448